metrics-monitoring/README.adoc

```diff
@@ -0,0 +1,225 @@
+WARNING: This content is generated by running npm --prefix .build run generate:attributes
+////
+
+
+:org-name: Application Services
+:community:
+:imagesdir: ./images
+:product-version: 1
+:product-long: OpenShift Streams for Apache Kafka
+:product: Streams for Apache Kafka
+:registry-product-long: OpenShift Service Registry
+:registry: Service Registry
+// Placeholder URL, when we get a HOST UI for the service we can put it here properly
+:service-url: https://console.redhat.com/application-services/streams/
+:registry-url: https://console.redhat.com/application-services/service-registry/
+:property-file-name: app-services.properties
+
+// Other upstream project names
+:samples-git-repo: https://github.com/redhat-developer/app-services-guides
+
+//URL components for cross refs
+:base-url: https://github.com/redhat-developer/app-services-guides/blob/main/
+:base-url-cli: https://github.com/redhat-developer/app-services-cli/tree/main/docs/
+:getting-started-url: getting-started/README.adoc
+:getting-started-service-registry-url: getting-started-service-registry/README.adoc
+:kafka-bin-scripts-url: kafka-bin-scripts/README.adoc
+:kafkacat-url: kafkacat/README.adoc
+:quarkus-url: quarkus/README.adoc
+:quarkus-service-registry-url: quarkus-service-registry/README.adoc
+:rhoas-cli-url: rhoas-cli/README.adoc
+:rhoas-cli-kafka-url: rhoas-cli-kafka/README.adoc
+:rhoas-cli-service-registry-url: rhoas-cli-service-registry/README.adoc
+:rhoas-cli-ref-url: commands
+:topic-config-url: topic-configuration/README.adoc
+:consumer-config-url: consumer-configuration/README.adoc
+:service-binding-url: service-discovery/README.adoc
+:access-mgmt-url: access-mgmt/README.adoc
+:access-mgmt-service-registry-url: access-mgmt-service-registry/README.adoc
+
+////
+END GENERATED ATTRIBUTES
+////
+
+[id="chap-monitoring-metrics"]
+= Monitoring metrics in {product-long}
+ifdef::context[:parent-context: {context}]
+:context: monitoring-metrics
+
+// Purpose statement for the assembly
+[role="_abstract"]
+As a developer or administrator, you can view metrics in {product} to visualize the performance and data usage for Kafka instances and topics that you have access to. You can view metrics directly in the {product} web console, or use the metrics API endpoint provided by {product} to import the data into your own metrics monitoring tool, such as Prometheus.
+
+//Additional line break to resolve mod docs generation error, not sure why. Leaving for now. (Stetson, 20 May 2021)
+
+[id="ref-supported-metrics_{context}"]
+== Supported metrics in {product}
+
+{product} supports the following metrics for Kafka instances and topics. In the {product} web console, a subset of these metrics is displayed in the *Dashboard* page of each Kafka instance.
+
+Broker metrics::
++
+--
+* `kafka_broker_quota_softlimitbytes`: Maximum amount of broker storage before producers are throttled
+
+* `kafka_broker_quota_totalstorageusedbytes`: Total storage used by the broker
```

Comment:
It's not a docs comment, but it's the first time I've really see how we're exposing metrics to users of the service... The naming of these could be problematic; they are not Kafka metrics. AFAIK they come from the [quota plugin](https://github.com/strimzi/kafka-quotas-plugin/blob/main/src/main/java/io/strimzi/kafka/quotas/StaticQuotaCallback.java#L160), where the name makes clear that they're a Strimzi, rather than Kafka thing. IMHO it is at least confusing to give these the `kafka_` prefix when they don't come from Kafka. It doesn't have to use a `strimzi` prefix, but perhaps we could use some other prefix. 

Slightly more generally, if there's a pattern to how we're using `:` in names it's not clear to me: Why is it `kafka_topic:kafka_topic_partitions...` but `kafka_broker_quota_...`?

@pmuir have we really thought about these names?

Reply:
@tombentley Ack. Yes, agreed that that's a convo outside the scope of this PR and one worth having. We'll move forward with whatever naming is currently a reality. Once they are updated, anyone is welcome to update the metrics names accordingly in the docs at that point. Thanks.

Reply:
OK, thanks. It's worth thinking about because this is API and people will build their own reporting and alerting around these things.

Reply:
@tombentley can you raise on the mas-devel list? I'm also not overly happy with the names of the metrics (they are hard to follow) but you probably have more insight into what they should be than I do.

Reply:
I also found the list of metrics difficult to grok, especially when they all appear in a long list. @sterobin you might consider switching the order and listing the definition first (since that's the part that most readers care about the most), and then the technical name of the metric.

So, for example instead of this:

```
* `kafka_broker_quota_softlimitbytes`: Maximum amount of broker storage before producers are throttled
```
Do something like this:

```
* Maximum amount of broker storage before producers are throttled (`kafka_broker_quota_softlimitbytes`) 
```

Reply:
Right, so the reason I listed the metrics like this is that this is how users will be keying them when they look them up. Meaning, they go to Prometheus or wherever and then open the list of available metrics that they can choose from to set up their metrics dashboard. So these are the metrics they'll need to choose from and will need to know what each means. So they're basically metrics entries that correlate with what they see in Prometheus. If I put definitions first, it will bury the things that they're comparing against for the information.

Regarding the confusingness of the metric names, that's certainly an issue, but a separate one. The names are definitely system-centered and not user-centered. That was Tom's

Reply:
`Slightly more generally, if there's a pattern to how we're using : in names it's not clear to me: Why is it kafka_topic:kafka_topic_partitions... but kafka_broker_quota_...?`

@tombentley I would expect any recording rules to follow the convention in https://prometheus.io/docs/practices/rules/#naming-and-aggregation